packages/core/src/mesh/SkinnedMeshRenderer.ts

```diff
@@ -391,4 +321,44 @@ export class SkinnedMeshRenderer extends MeshRenderer {
   private _onLocalBoundsChanged(): void {
     this._dirtyUpdateFlag |= RendererUpdateFlags.WorldVolume;
   }
+
+  private _getEntityHierarchyPath(rootEntity: Entity, searchEntity: Entity, inversePath: number[]): boolean {
+    inversePath.length = 0;
+    let entity = searchEntity;
+    while (entity !== rootEntity) {
+      const parent = entity.parent;
+      if (parent) {
+        inversePath.push(entity.siblingIndex);
+      } else {
+        return false;
+      }
+      entity = parent;
+    }
+    return true;
+  }
+
+  /**
+   * @internal
+   */
+  private _getEntityByHierarchyPath(rootEntity: Entity, inversePath: number[]): Entity {
+    let sprite = rootEntity;
+    for (let i = inversePath.length - 1; i >= 0; i--) {
+      sprite = sprite.children[inversePath[i]];
+    }
+    return sprite;
+  }
+
+  private _skin: Skin;
+
+  /**
+   * @deprecated
+   * Skin Object.
+   */
+  get skin(): Skin {
+    return this._skin;
+  }
+
+  set skin(value: Skin) {
+    this._skin = value;
```

Comment:
you should set renderer's `bones` and `rootBond` if you want adapter deprecated method